src/Adapter/MSTest.CoreAdapter/ObjectModel/UnitTestElement.cs

```diff
@@ -91,6 +96,7 @@ internal TestCase ToTestCase()
 
             TestCase testCase = new TestCase(fullName, TestAdapter.Constants.ExecutorUri, this.TestMethod.AssemblyName);
             testCase.DisplayName = this.TestMethod.Name;
```

Comment:
This line should be removed and should be clubbed with below line. Something like:
testcase.DisplayName = string.isNotNullorempty(this.DisplayName)? this.DisplayName : this.TestMethod.Name

Reply:
>clubbed 
that's a new one for me 😊

np, will do

Reply:
It's done